nsv13/code/modules/overmap/ai-skynet.dm

```diff
@@ -1228,12 +1232,17 @@ Seek a ship thich we'll station ourselves around
 					fire_weapon(ship, FIRE_MODE_GAUSS)
 					break
 		fire_mode = new_firemode
-		if(will_use_shot) //Don't penalise them for weapons that are designed to be spammed.
+		if(uses_main_shot) //Don't penalise them for weapons that are designed to be spammed.
 			shots_left --
+		else
+			light_shots_left --
 		fire_weapon(target, new_firemode, ai_aim=TRUE)
 		next_firetime = world.time + (1 SECONDS) + (fire_delay*2)
 		handle_cloak(CLOAK_TEMPORARY_LOSS)
 
+/obj/structure/overmap/proc/reload_light_weapon()
+	light_shots_left = initial(light_shots_left)
```

Comment:
You can use `addtimer(VARSET` instead of `addtimer(CALLBACK` if you want to avoid making one line procs. Alternatively you could use spawn(150) if you wanted to save overhead